app/services/adviser/sign_up.rb

```diff
@@ -0,0 +1,49 @@
+class Adviser::SignUp
+  class AdviserSignUpUnavailableError < RuntimeError; end
+
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+
+  attr_reader :application_form
+  attribute :preferred_teaching_subject
+
+  validates :preferred_teaching_subject, inclusion: { in: :teaching_subject_names, allow_blank: false }
+
+  def initialize(application_form, *args)
+    @application_form = Adviser::ApplicationFormValidations.new(application_form)
+
+    super(*args)
+  end
+
+  def save
+    raise AdviserSignUpUnavailableError unless available?
+
+    return false if invalid?
+
+    AdviserSignUpWorker.perform_async(application_form.id, preferred_teaching_subject_id)
+
+    true
+  end
+
+  def available?
+    feature_active? && application_form.valid?
+  end
+
+  def teaching_subjects
+    @teaching_subjects ||= GetIntoTeachingApiClient::LookupItemsApi.new.get_teaching_subjects
+  end
```

Comment:
Do these need to be public methods?

Reply:
Yeah, we check for `available?` to determine if we should present the sign up option to the candidate and the `teaching_subjects` are used as select options.